File: src/tests/_assets/framework_apps/javascript/jest_suite/sample.test.js
/**
 * Sample Jest tests for testing debugging capabilities.
 */

describe('Basic Tests', () => {
  test('simple assertion', () => {
    const x = 10;  //:bp.test.x:
    const y = 20;  //:bp.test.y:
    const result = x + y;  //:bp.test.result:
    expect(result).toBe(30);
  });

  test('calculation', () => {
    const a = 5;  //:bp.calc.a:
    const b = 3;  //:bp.calc.b:
    const product = a * b;  //:bp.calc.product:
    const total = a + b;  //:bp.calc.total:
    expect(product).toBe(15);
    expect(total).toBe(8);
  });

  test('variable types', () => {
    const integerVar = 42;  //:bp.vars.integer:
    const stringVar = "test";  //:bp.vars.string:
    const listVar = [1, 2, 3, 4, 5];  //:bp.vars.list:
    const dictVar = { key1: "value1", key2: "value2" };  //:bp.vars.dict:
    const nestedDict = {  //:bp.vars.nested:
      level1: { level2: { level3: "deep_value" } }
    };

    expect(integerVar).toBe(42);
    expect(stringVar).toBe("test");
    expect(listVar).toHaveLength(5);
    expect(dictVar.key1).toBe("value1");
    expect(nestedDict.level1.level2.level3).toBe("deep_value");
  });
});

describe('Async Tests', () => {
  test('async operation', async () => {
    const value = 100;  //:bp.async.value:
    const result = await Promise.resolve(value * 2);  //:bp.async.result:
    expect(result).toBe(200);
  });
});

describe('Test Class', () => {
  test('class method', () => {
    const x = 15;  //:bp.class.x:
    const y = 25;  //:bp.class.y:
    const total = x + y;  //:bp.class.total:
    expect(total).toBe(40);
  });
});

File: src/tests/_assets/framework_apps/java/junit_suite/src/test/java/com/example/test/SampleTest.java
package com.example.test;

import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.DisplayName;
import static org.junit.jupiter.api.Assertions.*;

import java.util.HashMap;
import java.util.Map;

public class SampleTest {

    @Test
    @DisplayName("Simple assertion test")
    public void testSimpleAssertion() {
        int x = 10;  //:bp.test.x:
        int y = 20;  //:bp.test.y:
        int result = x + y;  //:bp.test.result:
        assertEquals(30, result);
    }

    @Test
    @DisplayName("Calculation test")
    public void testCalculation() {
        int a = 5;  //:bp.calc.a:
        int b = 3;  //:bp.calc.b:
        int product = a * b;  //:bp.calc.product:
        int total = a + b;  //:bp.calc.total:
        assertEquals(15, product);
        assertEquals(8, total);
    }

    @Test
    @DisplayName("Variable types test")
    public void testVariableTypes() {
        int integerVar = 42;  //:bp.vars.integer:
        String stringVar = "test";  //:bp.vars.string:
        int[] listVar = {1, 2, 3, 4, 5};  //:bp.vars.list:

        Map<String, String> dictVar = new HashMap<>();  //:bp.vars.dict:
        dictVar.put("key1", "value1");
        dictVar.put("key2", "value2");

        Map<String, Object> nestedDict = new HashMap<>();  //:bp.vars.nested:
        Map<String, Object> level1 = new HashMap<>();
        Map<String, Object> level2 = new HashMap<>();
        level2.put("level3", "deep_value");
        level1.put("level2", level2);
        nestedDict.put("level1", level1);

        assertEquals(42, integerVar);
        assertEquals("test", stringVar);
        assertEquals(5, listVar.length);
        assertEquals("value1", dictVar.get("key1"));

        @SuppressWarnings("unchecked")
        Map<String, Object> level1Check = (Map<String, Object>) nestedDict.get("level1");
        @SuppressWarnings("unchecked")
        Map<String, Object> level2Check = (Map<String, Object>) level1Check.get("level2");
        assertEquals("deep_value", level2Check.get("level3"));
    }

    @Test
    @DisplayName("Exception test")
    public void testException() {
        String value = "test";  //:bp.exception.value:
        assertThrows(NumberFormatException.class, () -> {
            Integer.parseInt(value);  //:bp.exception.parse:
        });
    }

    @Test
    @DisplayName("Class method test")
    public void testClassMethod() {
        int x = 15;  //:bp.class.x:
        int y = 25;  //:bp.class.y:
        int total = x + y;  //:bp.class.total:
        assertEquals(40, total);
    }
}
